GLOSSARY.md

```diff
@@ -0,0 +1,27 @@
+# Glossary
+
+This document gathers the canonical nomenclature of the Lightning concepts used by the LDK contributors. Note, they may diverge from the BOLT specs or the ones used by the wider Lightning community, the test of time has revealed that it can be hard to agree on terms employed. The following terms are employed with a best effort across the codebase aiming to reduce both code reviewers and library users confusion.
+
+### Holder/Counterparty
+
+Inspired by financial cryptography, the holder is the entity beneficiary of a right. In LDK context, it designates the entity who is served by the implementation in its channel operations. The counterparty is the entity participating in the channel operation as a peer, but of whom the interests are not of concern by the implementation. Those roles are fixed for the lifetime of a channel and doesn't change in fonction of the state (message, blocks, ...) processed.
+
+Used across the channel-management subsystem (`Channel`).
+
+### Broadcaster/Countersignatory
+
+In Lightning, states are symmetrics but punishment is asymmetric, which forces channel parties to hold different commitment transactions. At transaction construction, the broadcaster designates the entity which has the unilateral capability to broadcast the tranaction. The countersignatory designates the entity providing signatures for the broadcastable transaction and thus verifying its encodes the off-chain states. Those roles are dynamic for the lifetime of a channel and change during the phases of a channel update.
```

Comment:
s/symmetrics/symmetric/
s/its/it/

also, I would not say the the role is dynamic.  it's just associated with a commitment tx.  i.e. at any point in time, there should be two "latest" commitment txs that have been processed by each party's implementation, one where the holder is the broadcaster and counterparty is countersignatory, and one where the holder is the contersignatory and the counterparty is the broadcaster.

Reply:
Same comment as above about `designates`, suggest replacement with `is`

@devrandom, I like the clarification and pulling in the use of `holder` and `counterparty`.  It helps cement the concepts in the prior entry.

Reply:
Done.